src/wifi/model/dmg-wifi-scheduler.h

```diff
@@ -0,0 +1,324 @@
+   * \param newStartTime The new starting time of the allocation.
+   * \param newDuration The new duration of the allocation.
+   */
+  void ModifyAllocation (AllocationID allocationId, uint8_t sourceAid, uint8_t destAid, 
+                         uint32_t newStartTime, uint16_t newDuration);
+  /**
+   * Creates a broadcast CBAP allocation field.
+   * \param staticAllocation Is the allocation static.
+   * \param allocationStart The starting time of the allocation.
+   * \param blockDuration The duration of the allocation.
+   * \return The broadcast CBAP allocation field.
+   */
+  AllocationFieldList GetBroadcastCbapAllocation (bool staticAllocation, uint32_t allocationStart, uint32_t blockDuration);
+  /**
+   * Cleanup non-static allocations.
+   */
+  void CleanupAllocations (void);
+  /**
+   * Allocate entire DTI as CBAP if no requested allocations are granted.
+   */
+  void AddBroadcastCbap (void);
+
+  /* Channel Access Period */
+  ChannelAccessPeriod m_currentAccessPeriod;   //!< The type of the current channel access period.
+  Time m_biDuration;                           //!< The length of the BI period.
+  Time m_atiDuration;                          //!< The length of the ATI period.
+  Time m_dtiDuration;                          //!< The length of the DTI period.
+  Time m_currentAccessPeriodStartTime;         //!< The start time of the current channel acess period.
+
+  /* Allocation */
+  typedef struct {
+    uint8_t sourceAid;
+    Mac48Address sourceAddr;
+    DmgTspecElement dmgTspec;
+  } AddtsRequest;
+  typedef std::vector<AddtsRequest> AddtsRequestList;
+  typedef AddtsRequestList::const_iterator AddtsRequestListCI;
+  AddtsRequestList m_receivedAddtsRequests;    //!< The list containing the ADDTS requests received during the DTI.
+
+  /* An allocation is uniquely identified by the tuple: Allocation ID, Source AID, Destination AID (802.11ad 10.4). */
+  typedef std::tuple<AllocationID, uint8_t, uint8_t> UniqueIdentifier;
+  typedef std::map<UniqueIdentifier, AddtsRequest> AllocatedRequestMap;
+  typedef AllocatedRequestMap::iterator AllocatedRequestMapI;
+  AllocatedRequestMap m_allocatedAddtsRequestMap; //!< The map containing the allocated ADDTS requests with their original allocation parameters.
+
+  uint32_t m_minBroadcastCbapDuration;          //!< The minimum duration of a broadcast CBAP to be present in the DTI.
+  uint32_t m_interAllocationDistance;           //!< The distance between two allocations to be used as broadcast CBAP.
+  bool m_isAddtsAccepted;                       //!< An ADDTS request received in the last DTI has been accepted.
+  bool m_isAllocationModified;                  //!< An allocation has been successfully modified.
+  bool m_isNonStatic;                           //!< A non-static allocation has been served in the last DTI.
+  bool m_isDeltsReceived;                       //!< A DELTS request has been received in the last DTI. 
```

Comment:
Is it "safe" to use a class variable for a single examined ADDTS, DELTS or non-static allocation? What if we receive more than one? Is the code related still valid? We can discuss it in a call, I put it here just as a reminder (I looked also at the related code ofc)

Reply:
We forget to discuss this. 
Ideally I just need one DELTS, new or modification ADDTS request to trigger the AddBroadcastCbapAllocations method, because something changed in the addtsAllocationList. 
Having different booleans can help with identifying  what exactly happend in the last DTI.
Let's discuss offline in case

Reply:
But I mean, once one variable changes its value (from false to true for example) in one BI, then it stays true for the entire duration of the simulation or does it refresh at the beginning of every BI? 
My point is, let's be sure that everything is updated correctly :)

Reply:
Yes they are updated accordingly after the method AddBraodcastCbapAllocations is called.